aws/resource_aws_iam_service_linked_role.go

```diff
@@ -0,0 +1,169 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsIamServiceLinkedRole() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsIamServiceLinkedRoleCreate,
+		Read:   resourceAwsIamServiceLinkedRoleRead,
+		Delete: resourceAwsIamServiceLinkedRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"aws_service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, es []error) {
+					value := v.(string)
+					if !strings.HasSuffix(value, ".amazonaws.com") {
+						es = append(es, fmt.Errorf(
+							"%q must be a service URL e.g. elasticbeanstalk.amazonaws.com", k))
+					}
+					return
+				},
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"path": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"create_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"unique_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsIamServiceLinkedRoleCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).iamconn
+
+	serviceName := d.Get("aws_service_name").(string)
+
+	params := &iam.CreateServiceLinkedRoleInput{
+		AWSServiceName: aws.String(serviceName),
+	}
+
+	_, err := conn.CreateServiceLinkedRole(params)
+
+	if err != nil {
+		return fmt.Errorf("Error creating service-linked role with name %s", serviceName)
+	}
+
+	d.SetId(serviceName)
```

Comment:
We should set this to the role ARN so we can utilize `GetRole` instead of the unsafe `ListRoles` (you can have multiple service linked roles for the same service when created with `CustomSuffix`)

Reply:
@bflad Not sure if I'm just missing something, but it doesn't look like you can use the ARN with `GetRole`, [just the role name](https://docs.aws.amazon.com/IAM/latest/APIReference/API_GetRole.html). Should I use the role name as the `Id`? That doesn't seem right either, but that is what `aws_iam_role` does. Should I do what you mention for the `DeleteServiceLinkedRole` call and just parse the role name out of the ARN?